pkg/db/subscription.go

```diff
@@ -79,7 +79,9 @@ func (s *DBSubscription[ValueType]) poll() {
 	// Repeatedly query page by page until no more results
 	for {
 		results, lastID, err := s.query(s.ctx, s.lastSeenID, s.options.NumRows)
-		if err != nil {
+		if s.ctx.Err() != nil {
+			break
+		} else if err != nil {
```

Comment:
It's pretty benign, but whenever I look at logs for the tests, they always end with entries saying `Error querying for DB subscription`. I assume this is happening during shutdown. This change was meant to silence those logs, but it didn't silence them - I suspect we're shutting things down in the wrong order. I can clean up the logs at a future date

Reply:
I do have a PR waiting in the wings that should simplify our shutdown flow. Not sure if it fixes this particular issue, but we can check after.